### PR TITLE
Add new capability for selecting a specific kernel index

### PR DESCRIPTION
**Pull request by afanfa**
Hi all,
this patch is meant to let the user choose a specific kernel identified by its index in the Master Solution Library.
Comments and suggestions are more than welcome.

Alessandro

### CODE REVIEW

**Comment by sdquiring** [comment]
First, we should be consistent in calling this a 'solution' index, not a 'kernel' index, since a solution can involve multiple kernels being invoked (for GSU).

Second, if this environment variable is set, then Tensile will only support one particular transpose, and one particular data type.  It may also have other restrictions in terms of problem/stride size since some solutions may have such limitations.  @zaliu I'm not sure if there's a need to communicate this better?

Finally, this would be best implemented in `MasterSolutionLibrary`: https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp#L87 as the the `solutions` member is a map from SolutionIndex to Solution.  We just need to check the predicates and then either return the solution or `nullptr` if the predicates fail.

**Comment by zaliu** [comment]
> @zaliu I'm not sure if there's a need to communicate this better?
Your comments are clear enough.  I'm sure @afanfa will communicate directly with us if clarifications are needed.  Thanks!

**Comment by sdquiring** [comment]
Sorry, by communicate I mean by means of a comment in the code or a printed warning when this environment variable is active.

**Comment by afanfa** [comment]
Thanks for the quick review. I will implement this functionality in MasterSolutionLibrary as suggested by @sdquiring with a better naming.

File: Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp [hunk 1]
```diff
@@ -84,7 +84,17 @@ namespace Tensile
...
         {
-            return library->findBestSolution(problem, hardware, fitness);
+            const int solution_index = Debug::Instance().getSolutionIndex();
+
+            if(solution_index >= 0)
+            {
+                std::cout << "Tensile will use solution index: " << solution_index <<std::endl;
+                std::cout << "Warning: Tensile will only work for a particular transpose and data type." << std::endl;
+                std::cout << "Set TENSILE_SOLUTION_INDEX to a negative number to restore the default behavior." << std::endl;
+                return solutions.at(solution_index);
```
Comment (sdquiring) [inline Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp:94]: You still need to check the predicates. `(*solution->problemPredicate)(problem)` and `(*solution->hardwarePredicate)(hardware)` must both return `true` or we should return `nullptr` from here.

**Review by sdquiring — COMMENTED** [review]
Great, thanks for implementing the changes.  One more quick note though.

**Review by sdquiring — APPROVED** [review]
Great! Ready to merge once CI passes.

**Comment by afanfa** [comment]
Something didn't work with my rebase.

**Pull request merged**